src/site/content/en/fast/efficiently-load-third-party-javascript/index.md

```diff
@@ -0,0 +1,174 @@
+
+### `defer`
+
+Scripts with the `defer` attribute execute after HTML parsing is completely finished, but before the [`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded) event. `defer` guarantees scripts will be executed in the order they appear in the HTML and will not block the parser.
+
+![Diagram of parser flow with a script with defer attribute](defer@2x.png)
+
+* Use `async` if it's important to have the script run earlier in the loading process. 
+
+* Use `defer` for less critical resources. A video player that's below-the-fold, for example.
+
+Using these attributes can significantly speed up page load. For example, [Telegraph recently deferred all of their scripts](https://medium.com/p/a0a1000be5#4123), including ads and analytics, and improved the First Ad Loaded metric by an average of four seconds.
+
+
+{% Aside %}
+Analytics scripts are usually loaded early to not miss any valuable analytics data. Fortunately, there are [patterns to initialize analytics lazily](https://philipwalton.com/articles/the-google-analytics-setup-i-use-on-every-site-i-build/) while retaining early page-load data.
+{% endAside %}
+
+## Establish early connections to required origins 
+
+Establishing connections takes significant time on slow networks. This is especially true for secure connections, which may involve DNS lookups, redirects, and several round trips to the final server that handles the user’s request. 
+
+You can save 100–500 ms by establishing early connections to important third-party origins.
+
+Two [`<link>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) types can help here:
+
+* `preconnect`
+
+* `dns-prefetch`
+
+### `preconnect`
+
+`<link rel="preconnect">` informs the browser that your page intends to establish a connection to another origin, and that you’d like the process to start as soon as possible. When the request for a resource from the preconnected origin is made, the download starts immediately. 
+
+```html
+<link rel="preconnect" href="https://cdn.example.com">
+```
+
+{% Aside 'caution' %}
+
+Only preconnect to critical origins you will use soon because if the connection isn’t used within 10 seconds the browser closes it. There’s a limit to the number of simultaneous connections a browser can handle (for example, Chrome can handle six), and unnecessary preconnecting can delay other important resources.
+
+{% endAside %}
+
+### `dns-prefetch`
+
+`<link rel="dns-prefetch>` handles a small subset of what is handled by `<link rel="preconnect">`.  Establishing a connection involves the DNS lookup and TCP handshake, and for secure origins, TLS negotiations. `dns-prefetch` instructs the browser to only resolve the DNS of a specific domain before it has been explicitly called.
+
+Because there’s a limit to the number of connections a browser can open, for less critical third-party origins use `<link rel=dns-prefetch>`. 
```

Comment:
```suggestion
Because there's a limit to the number of connections a browser can open, for less critical third-party origins use `<link rel=dns-prefetch>`. 
```
Curly apostrophe to straight.